storage/types.go

```diff
@@ -69,3 +71,9 @@ type StoreManager interface {
     GetTimeout() time.Duration
     Health() error
 }
+
+type RedisClient interface {
```

Comment:
Common interface to support both `redis.Client` and `redis.ClusterClient`.

Reply:
so, LOL, if you hit "resolve" (which, as the reviewer is my prerogative :laughing: ) I won't see your response ;)

right - so, I eventually got that - my problem with this approach is that, in due course, we will end up having to replicate the entire API of the redis client (or at least, whatever portion we need in our code) and this will greatly limit what we can do.

I understand what you're trying to do, and totally appreciate that the lack of abstract classes in Go greatly limit our choices here, but I would like to see if there is a less drastic way of achieving the same goal here.

Reply:
What other approach would you recommend @massenz ? (or can we add that as an issue for improvement?)